ecosystem/typescript/sdk_v2/src/internal/transaction.ts

```diff
@@ -32,3 +34,142 @@ export async function getGasPriceEstimation(args: { aptosConfig: AptosConfig })
   });
   return data;
 }
+
+export async function getTransactionByVersion(args: {
+  aptosConfig: AptosConfig;
+  txnVersion: AnyNumber;
+}): Promise<TransactionResponse> {
+  const { aptosConfig, txnVersion } = args;
+  const { data } = await getAptosFullNode<{}, TransactionResponse>({
+    aptosConfig,
+    originMethod: "getTransactionByVersion",
+    path: `transactions/by_version/${txnVersion}`,
+  });
+  return data;
+}
+
+export async function getTransactionByHash(args: {
+  aptosConfig: AptosConfig;
+  txnHash: HexInput;
+}): Promise<TransactionResponse> {
+  const { aptosConfig, txnHash } = args;
+  const { data } = await getAptosFullNode<{}, TransactionResponse>({
+    aptosConfig,
+    path: `transactions/by_hash/${txnHash}`,
+    originMethod: "getTransactionByHash",
+  });
+  return data;
+}
+
+export async function isTransactionPending(args: { aptosConfig: AptosConfig; txnHash: HexInput }): Promise<boolean> {
+  const { aptosConfig, txnHash } = args;
+  try {
+    const transaction = await getTransactionByHash({ aptosConfig, txnHash });
+    return transaction.type === TransactionResponseType.Pending;
+  } catch (e: any) {
+    if (e?.status === 404) {
+      return true;
+    }
```

Comment:
Interesting, we should probably in the future check if the response is the proper error, not just the 404, since you could get a 404 from elsewhere.